### PR TITLE
AWS component

### PR DESCRIPTION
**Pull request by awarecan**
**Description:**

Add a new aws component, will use it to consolidate several aws related component/platforms. 

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#22240

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_components/aws.markdown [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+## {% linkable_title Configuration %}
+
+To use `aws` component and `notify` platform in your installation, add the following to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+aws:
+  credentials:
+    - name: My AWS Account
+      aws_access_key_id: !secure some-id
```
Comment (klaasnicolaas) [inline source/_components/aws.markdown]:
```suggestion
      aws_access_key_id: AWS_ID
```

File: source/_components/aws.markdown [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      aws_secret_access_key: !secure some-key
```
Comment (klaasnicolaas) [inline source/_components/aws.markdown]:
```suggestion
      aws_secret_access_key: ACCESS_KEY
```

File: source/_components/aws.markdown [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+`aws` component is using [botocore](https://botocore.amazonaws.com/v1/documentation/api/latest/index.html) underneath to communicate with Amazon Web Services, which is also used in [AWS Command Client Interface](https://aws.amazon.com/cli/) tool. Therefore, `aws` shared same credential and profiles with `awscli` tool. Please read [Configuring the AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html) to learn how to get access keys and security manage them in your local system.
+
+## {% linkable_title Configuration %}
+
+To use `aws` component and `notify` platform in your installation, add the following to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+aws:
+  credentials:
```
Comment (klaasnicolaas) [inline source/_components/aws.markdown:33]: missing: `region_name: us-east-1`?

File: source/_components/aws.markdown [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    - name: My AWS Account
+      aws_access_key_id: !secure some-id
+      aws_secret_access_key: !secure some-key
+  notify:
+    # use the first credential defined in aws component by default
+    - service: lambda
+      name: Lambda Notify
```
Comment (klaasnicolaas) [inline source/_components/aws.markdown]: You can remove line 40 because it is not an required variable.

**Review by klaasnicolaas — DISMISSED** [review]

File: source/_components/aws.markdown (hunk 3 above)
Comment (awarecan) [reply source/_components/aws.markdown:33]: No,`region_name` is not required for `credential`, but required for notify service. e.g. credential is region independent.

File: source/_components/aws.markdown (hunk 4 above)
Comment (awarecan) [reply source/_components/aws.markdown]: It is not required but recommend. It can be omitted if only one notify configured.

Comment (klaasnicolaas) [reply source/_components/aws.markdown]: only required items in the first example

Comment (awarecan) [reply source/_components/aws.markdown]: I have it changed anyway.

**Review by frenck — APPROVED** [review]
Thanks! 👍

**Pull request merged**